File: tscServer/src/models/appointment.ts
import mongoose from "mongoose";
import {userSchema} from "./user";

export const appointmentSchema = new mongoose.Schema({
    lastActive: {
        type: Date,
        required: true,
        default: Date.now(),
    },
    createdAt: {
        type: Date,
        required: true,
        default: Date.now(),
    },
    shouldGetDoneWithin: {
        type: Date,
        required: true,
    },
    isDone: {
        type: Boolean,
        required: true,
    },
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User', // This should match the model name of your User model
        required: true,
    },
    doctorId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Doctor', // This should match the model name of your Doctor model
        required: true,
    },
});

const Appointment = mongoose.model("Appointment", appointmentSchema);
export default Appointment

File: lib/models/appointment.dart
class Appointment {
  final String id;
  final String userId;
  final String name;
  final String appointmentTime;
  final int serialNumber;
  final bool start_consultation_request_by_doctor;
  final String image_url;

  Appointment({
    required this.id,
    required this.userId,
    required this.name,
    required this.appointmentTime,
    required this.serialNumber,
    required this.start_consultation_request_by_doctor,
    required this.image_url,
  });
}
